client/my-sites/controller.js

```diff
@@ -552,9 +552,24 @@ export function siteSelection( context, next ) {
 		// onSelectedSiteAvailable might render an error page about domain-only sites or redirect
 		// to wp-admin. In that case, don't continue handling the route.
 		dispatch( setSelectedSiteId( siteId ) );
-		if ( onSelectedSiteAvailable( context ) ) {
-			next();
-		}
+
+		const currentPlanSlug = getSitePlanSlug( getState(), siteId );
+		const shouldUpdateStateAfterUpgrade =
+			context.pathname.startsWith( '/plans/my-plan/trial-upgraded/' ) &&
+			[ PLAN_FREE, PLAN_JETPACK_FREE ].includes( currentPlanSlug );
+
+		// This will fetch the site and update the state after the plan is upgraded if the site is on the trial-upgraded flow.
+		const promise = shouldUpdateStateAfterUpgrade
+			? dispatch( requestSite( siteId ) ).catch( () => {
+					dispatch( removeSite( siteId ) );
```

Comment:
On second thought, I'm not 100% sure that this makes sense. What if the site is just there but the request just failed? Why would we remove the known site object if we can just keep it and use it for the time being?

Reply:
I was concerned that the user could navigate to another page and see the free plan instead of the performance plan. But I made several tests here, and I don't think it's necessary.